src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    private static final BulldozerProducer BULLDOZER_PRODUCER = new BulldozerProducer();
+    private static final ExcavatorProducer EXCAVATOR_PRODUCER = new ExcavatorProducer();
+    private static final TruckProducer TRUCK_PRODUCER = new TruckProducer();
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        switch (type.getSimpleName()) {
+            case "Bulldozer": {
+                return new ArrayList<>(BULLDOZER_PRODUCER.get());
+            }
+            case "Excavator": {
+                return new ArrayList<>(EXCAVATOR_PRODUCER.get());
+            }
+            case "Truck": {
+                return new ArrayList<>(TRUCK_PRODUCER.get());
+            }
+            default:
+                return new ArrayList<>();
```

Comment:
Also I'd recommend creating a map as a class field (using class as a key and producer as a value) and using this map instead of switch